Sources/Commands/SwiftPackageTool.swift

```diff
@@ -392,12 +395,29 @@ public class SwiftPackageTool: SwiftTool<PackageToolOptions> {
                 usage: "The name of the package to unpin"),
             to: { $0.pinOptions.packageName = $1 })
 
+        let shellParser = parser.add(subparser: PackageMode.generateShellScript.rawValue, overview: "")
+        binder.bind(
+            positional: shellParser.add(
+                positional: "shell-name", kind: Shell.self,
+                usage: "The name of the shell to generate scripts for"),
+            to: { $0.shell = $1 })
+
         binder.bind(
             parser: parser,
             to: { $0.mode = PackageMode(rawValue: $1)! })
     }
 }
 
+enum Shell: String, StringEnumArgument {
```

Comment:
This should belong in `ArgumentParser` and not in swift tools because we want to be able to generate completion scripts for any parser instance.